setup.py

```diff
@@ -23,6 +33,58 @@ def read_requirements(file_path):
 version = current_date if package_name == "torchao-nightly" else "0.1"
 
 
+def get_extensions():
+    debug_mode = os.getenv('DEBUG', '0') == '1'
+    if debug_mode:
+        print("Compiling in debug mode")
+
+    use_cuda = torch.cuda.is_available() and CUDA_HOME is not None
+    extension = CUDAExtension if use_cuda else CppExtension
+
+    extra_link_args = []
+    extra_compile_args = {
+        "cxx": [
+            "-O3" if not debug_mode else "-O0",
+            "-fdiagnostics-color=always",
+        ],
+        "nvcc": [
+            "-O3" if not debug_mode else "-O0",
+        ]
+    }
+    if debug_mode:
+        extra_compile_args["cxx"].append("-g")
```

Comment:
will likely need to `conda install cuda -c nvidia` in CI since I don't think nvcc will be available